app/controllers/darjeelink/short_links_controller.rb

```diff
@@ -60,8 +60,12 @@ def handle_create_or_update(error_redirect_path:)
       begin
         yield
       rescue ActiveRecord::RecordNotUnique
-        flash[:error] = "#{params[:shortened_path]} already used! Choose a different custom path"
-        return redirect_to(error_redirect_path)
+        flash[:error] = "#{short_link_params[:shortened_path]} already used! Another custom path was autogenerated for you"
```

Comment:
This is the endpoint used when responding to a user-using-a-webbrowser and I think the error being caught is for when a user provides the name of a short link explicitly, but that's already been used. 

I think the code in api_controller is what you'd need to change..

Altho actually I think it might be better to do this logic in the short_link.rb file - if we say that it is the responsibility for auto_generate_shortened_path to create a path which is valid and usable, then the rest of the app doesn't need to know anything about how we go about that.